File: 210201/BaekJoon_1244_스위치켜고끄기.py
switch_num =input()
switch_info=list(map(int ,input().split(' ')))
stu_num = int(input())
stu_info=[]

for _ in range(stu_num):
    stu= list(map(int,input().split(' ')))
    stu_info.append(stu)
for x in range(len(stu_info)):
    if stu_info[x][0]==1:
        for y in range(len(switch_info)):

            if ((y+1) % stu_info[x][1]) ==0:
                if switch_info[y] ==0 :
                    switch_info[y] =1
                else:
                    switch_info[y] =0
        print(switch_info)
    else:
        if switch_info[stu_info[x][1]-1] == 0:
            switch_info[stu_info[x][1]-1] = 1
        else:
            switch_info[stu_info[x][1]-1] = 0
        left= stu_info[x][1]-2
        right=stu_info[x][1]
        while(left >=0 and (right <= len(switch_info)-1) and switch_info[left]==switch_info[right]):
            if switch_info[left] == 0:
                switch_info[left] = 1
            else:
                switch_info[left] = 0

            if switch_info[right] == 0:
                switch_info[right] = 1
            else:
                switch_info[right] = 0
            left -= 1
            right += 1
cnt=1
for x in switch_info:
    print(x,end='')
    if cnt<len(switch_info):
        print(' ', end='')
    if(cnt%20==0):
        print('')
    cnt += 1

File: 210201/BaekJoon_1244_스위치켜고끄기.java
package day0201;

import java.util.Scanner;

public class hwalgo01_대전_5반_노선호 {

	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);

		int switchNum = sc.nextInt();
		boolean[] arr = new boolean[switchNum];
		for (int i = 0; i < arr.length; i++) {
			int temp = sc.nextInt();
			if (temp == 1) {
				arr[i] = true;
			} else {
				arr[i] = false;
			}
		}

		int stuNum = sc.nextInt();
		int[][] stuInfo = new int[stuNum][2];
		for (int i = 0; i < stuInfo.length; i++) {
			for (int j = 0; j < stuInfo[i].length; j++) {
				stuInfo[i][j] = sc.nextInt();
			}
		}

		for (int i = 0; i < stuInfo.length; i++) {
			if (stuInfo[i][0] == 1) {
				for (int j = 0; j < arr.length; j++) {
					if ((j + 1) % stuInfo[i][1] == 0) {
						arr[j] = (!arr[j]);
					}
				}

			} else {

				arr[stuInfo[i][1] - 1] = !(arr[stuInfo[i][1] - 1]);

				int left = stuInfo[i][1] - 2;
				int right = stuInfo[i][1];
			

				while (left >= 0 && (right <= switchNum - 1) && (arr[left] == arr[right])) {
					arr[left] = (!arr[left]);
					arr[right] = (!arr[right]);
					left--;
					right++;

				}

			}

		}
		int cnt=1;
		for (int i = 0; i < arr.length; i++) {
			
			
			if (arr[i] == true) 
				System.out.print(1);
			else
				System.out.print(0);
			if(cnt<arr.length) {
			System.out.print(" ");}
			
			if(cnt%20 ==0) {
				System.out.println();
			}
			cnt++;
			
		

	}
	

}}

